packages/cron/src/jobs/measureNftTimeToRetrievability.js

```diff
@@ -0,0 +1,278 @@
+)
+
+/**
+ * @typedef RetrieveLog
+ * @property {"retrieve"}   type
+ * @property {string}       image
+ * @property {URL}          url
+ * @property {number}       contentLength
+ * @property {Date}         startTime
+ * @property {Milliseconds} duration
+ */
+
+/**
+ * @typedef {import('nft.storage/dist/src/token').TokenInput} TokenInput
+ */
+
+/**
+ * @template {TokenInput} T
+ * @typedef {import('nft.storage').TokenType<T>} TokenType
+ */
+
+/**
+ * @typedef {(token: TokenInput) => Promise<Pick<TokenType<TokenInput>, 'ipnft'>>} StoreFunction
+ */
+
+/**
+ * @returns {AsyncIterable<File>}
+ */
+export function TestImages(count = 1) {
+  return (async function* () {
+    while (count--) {
+      yield new File([Uint8Array.from([1, 2, 3])], 'image.png', {
+        type: 'image/png',
+      })
+    }
+  })()
+}
+
+/**
+ * @param  {...URL} urls
+ * @returns {AsyncIterable<Blob>}
+ */
+export function UrlImages(...urls) {
+  return (async function* () {
+    for (const url of urls) {
+      const resp = await fetch(url.toString())
+      yield resp.blob()
+    }
+  })()
+}
+
+/**
+ * @typedef HttpAuthorization
+ * @property {string} authorization
+ */
+
+/**
+ * Job that tests/measures steps
+ * * prepare a sample image
+ * * upload to nft.storage
+ * * retrieve image through ipfs gateway
+ * @template {import('nft.storage/dist/src/lib/interface').TokenInput} T
+ * @param {object} config
+ * @param {AsyncIterable<Blob>} config.images - images to upload/retrieve
+ * @param {StoreFunction} [config.store] - function to store nft
+ * @param {string} [config.url] - URL to nft.storage to measure
+ * @param {boolean} [config.logConfigAndExit] - if true, log config and exit
+ * @param {string} [config.metricsPushGateway] - Server to send metrics to. should be a https://github.com/prometheus/pushgateway
+ * @param {URL[]} config.gateways - IPFS Gateway to test retrieval from
+ * @param {import('../lib/log.js').LogFunction} config.log - logger
+ * @param {object} secrets
+ * @param {string} secrets.nftStorageToken - API Token for nft.storage
+ * @param {HttpAuthorization} secrets.metricsPushGatewayAuthorization - authorization header value for metricsPushGateway
+ */
+export async function measureNftTimeToRetrievability(config, secrets) {
+  if (config.logConfigAndExit) {
+    config.log('info', config)
+    return
+  }
+  const start = {
+    type: 'start',
+    job: 'measureNftTimeToRetrievability',
+    config,
+  }
+  config.log('info', start)
+  for await (const image of config.images) {
+    const imageId = Number(new Date()).toString()
+    const nft = {
+      image: /** @type {Blob} */ (image), // use image Blob as `image` field
+      name: 'Example NFT',
+      description: 'Example NFT used by nft.storage cron nft-ttr',
+    }
+    const client = new NFTStorage({
+      token: secrets.nftStorageToken,
+    })
+    const storeStartedDate = new Date()
+    const storeStartedAt = now()
```

Comment:
imho passing around an untyped int is more complexity than this because then whoever's dealing with the code later will have to remember the units instead of the compiler doing it for us.